### PR TITLE
Fix critical CSS generation failure due to invalid URLs

### LINKED ISSUE #29514: [Boost] Critical CSS generation failure due to invalid CSS URL
Source: p1678977893447429-slack-C016BBAFHHS

When attempting to parse an invalid URL, Critical CSS generation can fail with the following unhelpful error message:

> The [url] does not have any CSS in its external style sheet(s).

Here is an example faulty URL which can cause this issue: 

```
<link rel='stylesheet' id='fonts-css' href='https://example.com/@font-facefont-family:%20&#039;Custom&#039;;font-weight:%20400;font-style:%20normal;font-stretch:%20normal;font-display:%20swap;src:%20url(&#039;https://example.com/wp-content/themes/sample/assets/fonts/Custom-85.woff2&#039;)%20format(&#039;woff2&#039;);?ver=1678932203' media='all' />
````

This occurs even if there are other valid CSS URLs in the same HTML file. It appears that `collateCssFiles` is at fault; it catches errors at a per-page level, so if one CSS URL fails in a given page it stops trying.

### PR DESCRIPTION
**Pull request by haqadn**
Fixes Automattic/jetpack#29514

### CODE REVIEW

File: src/generate-critical-css.ts [hunk 1]
```diff
@@ -36,8 +36,12 @@ async function collateCssFiles(
 			// Convert relative URLs to absolute.
 			const relativeUrls = Object.keys( cssIncludes );
 			const absoluteIncludes = relativeUrls.reduce( ( set, relative ) => {
-				const absolute = new URL( relative, url ).toString();
-				set[ absolute ] = cssIncludes[ relative ];
+				try {
+					const absolute = new URL( relative, url ).toString();
+					set[ absolute ] = cssIncludes[ relative ];
+				} catch ( err ) {
+					// Ignore invalid URLs.
```
Comment (thingalon) [inline src/generate-critical-css.ts:43]: I feel this should log a warning that a URL is invalid, with the URL that is being rejected. It'll help us debug future issues. :)

**Review by thingalon — CHANGES_REQUESTED** [review]
Minor improvement, plus don't forget the linting whinges. :)

File: src/generate-critical-css.ts (hunk 1 above)
Comment (haqadn) [reply src/generate-critical-css.ts:43]: Log added. Do you mean disabling eslint because of the `console.log`, or am I missing some other lint whinings?

Comment (thingalon) [reply src/generate-critical-css.ts:43]: When I made this suggestion, I missed that this was inside the critical-css-gen lib, and my brain was in Hydra. Hydra definitely should log this message, but the critical-css-gen lib has no way to do that neatly.

For now a console.log will do.

But I would like to revisit the whole library's interface to make it easier to use, and easier to handle errors.

**Review by thingalon — APPROVED** [review]

**Pull request merged**